price_feeder/tasks_manager.py

```diff
@@ -127,6 +130,226 @@ def start_loop(self):
+
+        # get the last nonce
+        last_used_nonce = web3.eth.get_transaction_count(account_address)
+
+        if not task.pending_transactions:
+            task.pending_transactions = list()
+        else:
+
+            # semaphore to clear the queue of pending transactions
+            clear = False
+
+            # iterate over the pending transaction and update status in the queue
+            for tx in task.pending_transactions:
+
+                try:
+                    web3.eth.get_transaction(tx['hash'])
+                except exceptions.TransactionNotFound:
+
+                    # 4. STATUS: Dropped Tx
+
+                    # Transaction not exist anymore or dropped, permit to send new tx
+                    elapsed = datetime.datetime.now() - tx['timestamp']
+
+                    if tx['is_replacement']:
+                        action_label = 'Task :: {0} :: TX Replacement Dropped.'.format(task.task_name)
+                    else:
+                        action_label = 'Task :: {0} :: TX Dropped.'.format(task.task_name)
+
+                    log.info(
+                        "{0}"
+                        " Hash: [{1}] "
+                        " Price Oracle: [{2}] "
+                        " Price Last Time: [{3}] "
+                        " Price New: [{4}] "
+                        " Variation Oracle: [{5}] "
+                        " Variation Last Time: [{6}] "
+                        " Gas Price: [{7}] "
+                        " Nonce: [{8}] "
+                        " Elapsed: [{9}]".format(
+                            action_label,
+                            Web3.to_hex(tx['hash']),
+                            tx['price_oracle'],
+                            tx['price_last_time'],
+                            tx['price_new'],
+                            tx['variation_oracle'],
+                            tx['variation_last_time'],
+                            Web3.from_wei(tx['gas_price'], 'gwei'),
+                            tx['nonce'],
+                            elapsed.seconds
+                        )
+                    )
+
+                    continue
+
+                try:
+                    tx_rcp = web3.eth.get_transaction_receipt(tx['hash'])
+                except exceptions.TransactionNotFound:
+
+                    # 1. STATUS: Pending tx
+
+                    # timeout and permit to send again transaction
+                    elapsed = datetime.datetime.now() - tx['timestamp']
+
+                    if tx['is_replacement']:
+                        timeout_waiting = self.config['timeout_pending_replace_tx']
+                        label_timeout = 'Timeout TX Replacement!'
+                    else:
+                        timeout_waiting = self.config['timeout_pending_tx']
+                        label_timeout = 'Timeout TX!'
+
+                    timeout = datetime.timedelta(seconds=timeout_waiting)
+
+                    if elapsed > timeout:
```

Comment:
it this neccessary? If one of the txs is on timeout we are cleaning the whole list and eventually one of them will be confirmed

Reply:
yes and no, this is an extra checking in the case of some weird trouble that node keeping saying that is pending and really is not.